primap2/pm2io/_data_reading.py

```diff
@@ -1418,7 +1421,7 @@ def preferred_unit(entity: str, units: List[str], gwp_to_use: Optional[str]) ->
     >>> preferred_unit("CH4", ["kt CO2 / yr"], "AR4GWP100")
     'Gg CH4 / yr'
     >>> preferred_unit("CH4", ["kt CO2 / yr", "Mg CO2 / yr"], None)
-    'kt CO2 / yr'
+    'Gg CO2 / yr'
```

Comment:
But why should this be a better result than `kt CO2 / yr`? That seems very specialized to always prefer `Gg`, even if the input data are not ever given in Gg.

Reply:
Maybe Gg was the standard in primap1 and I got used to it. It's important however, that a default unit is given that is both used when converting from GWP weighted units and native units because else we will end up with different native units for the same entity. A way around this would be to convert the GWP weighted units to the present native unit in case it is present. Don't know how much work this is though.

Reply:
To convert to the first non-GWP unit instead of Gg the harmonize_units function needs a structural change. Currently it iterates over all entities. The entities for GWP weighted units and native units are different as the GWP is included in the entity name (e.g. "CH4 (SARGWP100)"). Thus to change it we would have to first group the entities by "basic entity" and then do the conversion. I can implement that but not in the next weeks as I'm busy with the PRIMAP-hist update

Reply:
What I don't understand is: Why are we changing the entity in `harmonize_units` at all? If we iterate over entities and treat GWP entities and mass entities as different entities, then we will never have a problem if they have non-matching units, right?